boards/x86/acrn/acrn.dts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <mem.h>
+
+#define DT_FLASH_SIZE	DT_SIZE_K(8192)
+#define DT_SRAM_SIZE	DT_SIZE_K(8192)
+
+#include <ia32.dtsi>
+
+/ {
+	model = "ACRN";
+	compatible = "intel,ia32";
+
+	aliases {
+		uart-0 = &uart0;
+		uart-1 = &uart1;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+	};
+};
+
+&uart0 {
+	status = "ok";
+	current-speed = <115200>;
+};
+
+&uart1 {
+	status = "ok";
+	current-speed = <115200>;
+};
+
+&flash0 {
+	reg = <0x100000 DT_FLASH_SIZE>;
+};
+
+&sram0 {
+	reg = <0x400000 DT_SRAM_SIZE>;
```

Comment:
This value I think is a legacy of QEMU's XIP emulation that got copypasted to some other x86 board setups.
I think we can just start SRAM from 0, and tell ACRN's loader to load from there. This will waste the least memory for the ACRN virtual partition.

Reply:
The build actually outputs to the flash region, not the SRAM region.  But to your point, I don't think we want to output to address 0, because x86 generally has MMIO between 640K and 1MB, which is why I presume the system is loaded at 1MB.

Reply:
> The build actually outputs to the flash region, not the SRAM region.

Ah I see. DT_SRAM_SIZE only seems to be a thing for XIP systems. Probably some cruftiness in how the x86 definitions were ported to DTS (which happened long after the fact).

> I don't think we want to output to address 0, because x86 generally has MMIO between 640K and 1MB, which is why I presume the system is loaded at 1MB.

Fair enough. We can re-visit if someone complains about too much RAM being used. I spent some time staring at https://wiki.osdev.org/Memory_Map_(x86) and couldn't convince myself that the first MB is always safe when the CPU is in 32-bit protected mode.